boa/src/builtins/nan/mod.rs

```diff
@@ -25,9 +27,14 @@ impl NaN {
 
     /// Initialize the `NaN` property on the global object.
     #[inline]
-    pub(crate) fn init(_interpreter: &mut Interpreter) -> (&'static str, Value) {
+    pub(crate) fn init(interpreter: &mut Interpreter) {
         let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
 
-        (Self::NAME, Value::from(f64::NAN))
+        let mut global = interpreter.global().as_object_mut().expect("Expect object");
+        global.borrow_mut().insert_property(
+            Self::NAME,
+            Value::from(f64::NAN),
+            Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
```

Comment:
`NaN` should be `Attribute::READONLY | Attribute::NON_ENUMERABLE | Attribute::PERMANENT`